File: CodeChef/Temple_Land.py

s = int(input())

for _ in range(s):
    n = int(input())

    h = list(map(int, input().split()))

    if(n % 2 == 0):
        print("no")
        continue

    start = 0
    end = n-1
    count = 1
    flag = False

    while(start <= end):

        if(h[start] != count or h[end] != count):
            flag = True
            break
        else:
            start += 1
            end -= 1
            count += 1
    
    if flag == False:
        print("yes")
    else:
        print("no")

File: CodeChef/Temple_Land.cpp
#include<bits/stdc++.h>
using namespace std;

int main()
{
    int t;
    cin >> t;

    while (t--) 
    {
        int n;
        cin >> n;
        int a[n];

        for (int i = 0; i < n; i++)
            cin >> a[i];
        
        if(n % 2 == 0)
        {
            cout << "no" <<endl;
            continue;
        }
        
        int start = 0, end = n-1, count = 1;
        bool flag = false;
        while(start <= end)
        {
            if((a[start] != count) || (a[end] != count))
            {
                flag = true;
                break;
            }
            else
            {
                start++;
                end--;
                count++;
            }
        }

        if(flag == false)
            cout << "yes" << endl;
        else
            cout << "no" << endl;

    }

}